### PR TITLE
WFCORE-3464 initial changes to make build work on JDK10

### PR DESCRIPTION
**Pull request by ctomc**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -131,8 +131,8 @@
         <version.org.jboss.xnio>3.5.4.Final</version.org.jboss.xnio>
         <version.org.jboss.xnio.xnio-api>${version.org.jboss.xnio}</version.org.jboss.xnio.xnio-api>
         <version.org.jboss.xnio.xnio-nio>${version.org.jboss.xnio}</version.org.jboss.xnio.xnio-nio>
-        <version.org.jmockit>1.33</version.org.jmockit>
-        <version.org.mockito>2.8.47</version.org.mockito>
+        <version.org.jmockit>1.37</version.org.jmockit>
```
[the remark this answers is not in the document]
Comment (ctomc) [reply pom.xml]: not really, we need to update *mock* to make it work on newer jdk

[the remark this answers is not in the document]
Comment (dmlloyd) [reply pom.xml]: Sure but that way if something goes wrong, a bisect can tell whether it was the jmockit/mockito change that broke it, or the stuff that comes after.

[the remark this answers is not in the document]
Comment (bstansberry) [reply pom.xml]: +1

[the remark this answers is not in the document]
Comment (ctomc) [reply pom.xml]: I've split it in few commits now.

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
I removed the Ready for Merge label because when I put this on master-ignore with a bunch of others, I got consistent failures a la https://ci.wildfly.org/viewLog.html?buildId=85017&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreMasterIgnoreIntegration

Removing this one from the set eliminated the problem. TBH I don't see why my 'retest this please' above didn't produce the problem; perhaps it's only in combo with the other PRs. Those are now merged, so we'll see.

**Comment by bstansberry** [comment]
retest this please

**Comment by ctomc** [comment]
@bstansberry there is an issue in WildFly full that clashes with https://github.com/wildfly/wildfly-core/pull/2999/commits/acf9f3d1c5e28ef7f2a23fc56c16e1ecd1a80c78
I will send PR to full to add excludes.

**Comment by ctomc** [comment]
@bstansberry merging https://github.com/wildfly/wildfly/pull/10766 should fix also potential enforcer error.

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**